jack-and-the-beans/buddies-ios
# Bud 17 report user

### In this PR
* Report activities with full text area
* Report users with report modal
* When you report someone/something, you go back to your root view controller
* Reports persist to firebase

##### UI change
<img width="358" alt="Screen Shot 2019-04-01 at 10 26 49 AM" src="https://user-images.githubusercontent.com/7787605/55335222-bca87780-5468-11e9-8162-4d4ef189d339.png">

![image](https://user-images.githubusercontent.com/7787605/55335212-b914f080-5468-11e9-80bf-1799231d9218.png)

### TODO
- [ ] Rebase on master after #120 and #116 are done
- [ ] Drop both Pod commits and re-reinstall the pod to avoid ugly conflicts with master


### To test

* Report a user
    * Make sure a `user_report` document is created with correct IDs
    * Make sure recipient has blocker in `blocked_by` list
    * Make sure instigator has blocked in `blocked_users`
    * Make sure you return back to the root of your tab VC (i.e. discover/my activities/topics)
* Report an activity
    * Make sure a `activity_report` document is created with correct IDs
    * Make sure your `account` has the activity in its `blocked_activitites` array
    * Make sure you return back to root VC
